c++/src/kj/string.h

```diff
@@ -134,8 +134,10 @@ class StringPtr {
 
   inline constexpr bool operator==(decltype(nullptr)) const { return content.size() <= 1; }
 
-  inline bool operator==(const StringPtr& other) const;
-  inline bool operator< (const StringPtr& other) const;
+  inline bool operator==(const StringPtr& other) const { return content.first(content.size() - 1) == other.content.first(other.content.size() - 1); }
+  //The -1 is to ignore the null character
```

Comment:
This is an implementation comment, but you've placed it where a doc comment belongs. In theory, if we were auto-generating docs or something, the `operator==()` would end up documented as "The -1 is to ignore the null character", which doesn't make sense.

If you want to keep this comment, I think you should change `operator==` back to being defined out-of-line later in the file, as it was before this PR. You can then put this implementation comment inside the definition, where it won't be interpreted as documentation.

Same goes for `operator<` below.

Reply:
changed, thank you for the advice.